composer/algorithms/augmix/README.md

```diff
@@ -0,0 +1,108 @@
+# 🎨 AugMix
+
+[\[How to Use\]](#how-to-use) - [\[Suggested Hyperparameters\]](#suggested-hyperparameters) - [\[Technical Details\]](#technical-details) - [\[Attribution\]](#attribution)
+
+`Computer Vision`
+
+For each data sample, AugMix creates an _augmentation chain_ by sampling `depth` image augmentations from a set (e.g. translation, shear, contrast).
+It then applies these augmentations sequentially with randomly sampled intensity.
+This is repeated `width` times in parallel to create `width` different augmented images.
+The augmented images are then combined via a random convex combination to yield a single augmented image, which is in turn combined via a random convex combination sampled from a Beta(`alpha`, `alpha`) distribution with the original image.
+
+| ![AugMix](aug_mix.png) |
+|:--:
+|*An image of a turtle that undergoes three different augmentation chains, gets combined back together using a convex combination, and gets combined with the original image. [Figure 4 from Hendrycks et al. (2020)](https://arxiv.org/abs/1912.02781).*|
+
+## How to Use
+
+### Functional Interface
+
+TODO(MATTHEW): FIX
+
+```python
+def training_loop(model, train_loader):
+  opt = torch.optim.Adam(model.parameters())
+  loss_fn = F.cross_entropy
+  model.train()
+  
+  for epoch in range(num_epochs):
+      for X, y in train_loader:
+          y_hat = model(X)
+          loss = loss_fn(y_hat, y)
+          loss.backward()
+          opt.step()
+          opt.zero_grad()
+```
```

Comment:
```suggestion
```python
def make_augmented_dataloader() -> DataLoader:
    # use the augmix_image transform when creating the dataset
    train_dataset = torchvision.datasets.CIFAR10(root="datadir", train=True, download=True, transform=cf.augmix_image)
    
    train_dataloader = DataLoader(train_dataset, batch_size=1024)
    return train_dataloader
```
```

Reply:
While technically the functional form is composable, we discourage this because doing so either 1) requires wrapping it in a lambda, or 2) using the default param values

Reply:
@growlix Oh I see if that's the case then can this be updated to reflect the proper usage as well: https://colab.research.google.com/drive/1HIxLs61pyf0ln7MlnrGYvkNHq1uVbNWu#scrollTo=udTtW9JUmtKU

It's the tutorial for the functional API I got that example from here